libraries/helpers.rb

```diff
@@ -0,0 +1,22 @@
+module ChefAnalyticsCookbook
+  module Helpers
+    def api_fqdn_available?
+      return false if node['chef-analytics'].nil?
+      return false if node['chef-analytics']['api_fqdn'].nil?
+      !node['chef-analytics']['api_fqdn'].empty?
```

Comment:
`node['chef-analytics'] && node['chef-analytics']['api_fqdn'] && !node['chef-analytics']['api_fqdn'].empty?`

its a long expression, but until we can drop support for 2.1+2.2 and can use the new operator in 2.3 that is more idiomatic for protecting against method chaining trainwrecks.